link error on Windows

Hi,
Thanks for the great work.
I'm trying to build `thrift` which depends on  `@boost//:locale`, and `@boost//:numeric_conversion`.
This is my BUILD file, which works fine on Linux but I get link error on Windows.
```bazel
cmake(
    name = "thrift",
    cache_entries = {
        "CMAKE_BUILD_TYPE": "Release",
        "BUILD_COMPILER": "OFF",
        "BUILD_CPP": "ON",
        "BUILD_LIBRARIES": "ON",
        "BUILD_NODEJS": "OFF",
        "BUILD_PYTHON": "OFF",
        "BUILD_JAVASCRIPT": "OFF",
        "BUILD_C_GLIB": "OFF",
        "BUILD_JAVA": "OFF",
        "BUILD_TESTING": "OFF",
        "BUILD_TUTORIALS": "OFF",
        "WITH_HASKELL": "OFF",
    },
    copts = [
        "-Ilibs/exporters/jaeger/openssl/include",
    ],
    lib_source = "@com_github_thrift//:all_srcs",
    out_static_libs = [
        "libthrift.a",
        "libthriftz.a",
    ],
    visibility = ["//visibility:public"],
    deps = [
        ":openssl",
        ":openssl_header_only",
        "@boost//:locale",
        "@boost//:numeric_conversion",
    ],
)
```

Error
```
ERROR: C:/users/runneradmin/_bazel_runneradmin/cefevaif/external/boost/BUILD.bazel:357:11: Linking external/boost/atomic_sse_898a644474.dll failed: (Exit 1120): link.exe failed: error executing command C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/link.exe @bazel-out/x64_windows-fastbuild/bin/external/boost/atomic_sse_898a644474.dll-2.params
   Creating library bazel-out/x64_windows-fastbuild/bin/external/boost/atomic_sse.if.lib and object bazel-out/x64_windows-fastbuild/bin/external/boost/atomic_sse.if.exp
find_address_sse2.obj : error LNK2019: unresolved external symbol "unsigned __int64 __cdecl boost::atomics::detail::find_address_generic(void const volatile *,void const volatile * const *,unsigned __int64)" (?find_address_generic@detail@atomics@boost@@YA_KPEDXPEBQEDX_K@Z) referenced in function "unsigned __int64 __cdecl boost::atomics::detail::find_address_sse2(void const volatile *,void const volatile * const *,unsigned __int64)" (?find_address_sse2@detail@atomics@boost@@YA_KPEDXPEBQEDX_K@Z)
find_address_sse41.obj : error LNK2001: unresolved external symbol "unsigned __int64 __cdecl boost::atomics::detail::find_address_generic(void const volatile *,void const volatile * const *,unsigned __int64)" (?find_address_generic@detail@atomics@boost@@YA_KPEDXPEBQEDX_K@Z)
bazel-out\x64_windows-fastbuild\bin\external\boost\atomic_sse_898a644474.dll : fatal error LNK1120: 1 unresolved externals
ERROR: D:/a/opentelemetry-cpp/opentelemetry-cpp/exporters/jaeger/BUILD:145:6 Foreign Cc - CMake: Building thrift failed: (Exit 1120): link.exe failed: error executing command C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/link.exe @bazel-out/x64_windows-fastbuild/bin/external/boost/atomic_sse_898a644474.dll-2.params
```

I'd appreciate any ideas to get this fixed.

--- a/BUILD.boost
+++ b/BUILD.boost
@@ -324,6 +324,10 @@ cc_library(
         "-msse4.1",
         "-Iexternal/boost/libs/atomic/src",
     ],
+    linkstatic = select({
+        ":windows_x86_64": True,
+        "//conditions:default": False,
+    }),
     visibility = ["//visibility:private"],
     deps = BOOST_ATOMIC_DEPS,
 )

